### PR TITLE
Do not roll out new application package with pure deployment orchestr…

### PR DESCRIPTION
**Pull request by jonmv**
…ation changes

@mpolden please review.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/Application.java [hunk 1]
```diff
@@ -118,6 +121,20 @@ public Application(TenantAndApplicationId id, Instant createdAt, DeploymentSpec
         return versions;
     }
 
+    /** Returns the currently deployed versions of the application */
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/Application.java:124]:
```suggestion
    /** Returns all current application version which change deployments, relative to the previous one(s) */
```

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/Application.java [hunk 2]
```diff
@@ -118,6 +121,20 @@ public Application(TenantAndApplicationId id, Instant createdAt, DeploymentSpec
...
+    public Collection<ApplicationVersion> deployableVersions(boolean ascending) {
+        Deque<ApplicationVersion> versions = new ArrayDeque<>();
+        String previousHash = "";
+        for (ApplicationVersion version : versions()) {
+            if (previousHash.equals(previousHash = version.bundleHash().orElse("")) && version.bundleHash().isPresent())
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/Application.java:129]: Consider putting assignment on separate line. I don't think I've ever seen an equals call that also assigns a variable. 😆

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentContext.java [hunk 3]
```diff
@@ -254,14 +260,29 @@ public DeploymentContext submit(ApplicationPackage applicationPackage) {
...
+        return submit(applicationPackage, sourceRevision, salt.getAndIncrement());
+    }
+
+    /** Submit given application package for deployment */
+    public DeploymentContext submit(ApplicationPackage applicationPackage, Optional<SourceRevision> sourceRevision, long salt) {
+        var buffer = new ByteArrayOutputStream();
+        ZipStreamReader.transferAndWrite(buffer,
+                                         new ByteArrayInputStream(applicationPackage.zippedContent()),
+                                         "salt",
+                                         new byte[]{ (byte) (salt >> 56), (byte) (salt >> 48), (byte) (salt >> 40), (byte) (salt >> 32), (byte) (salt >> 24), (byte) (salt >> 16), (byte) (salt >> 8), (byte) salt });
```
Comment (mpolden) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentContext.java:279]: 🙂

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/Application.java (hunk 2 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/Application.java:129]: I've seen it multiple places in the standard library 😬 But sure. It's just what felt natural writing. Maybe not the easiest thing to parse. though ... 😀

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentContext.java (hunk 3 above)
Comment (jonmv) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentContext.java:279]: 🙈

**Review by mpolden — APPROVED** [review]

**Pull request merged**